In [80]:
import pyAgrum as gum
import pyAgrum.lib.notebook as gnb

In [58]:
bn=gum.BayesNet('SmokerTreatment')

In [59]:
smoker = bn.add('smoker',2)
treatment = bn.add('treatment',2)
dead = bn.add('dead',2)

In [60]:
bn.addArc(smoker,treatment)
bn.addArc(smoker,dead)
bn.addArc(treatment,dead)

In [75]:
bn.cpt(smoker)[:]=[0.7,0.3]
bn.cpt(treatment)[:]=[[0.8,0.2],[0.45,0.55]]
print(bn.cpt(dead).var_names)
bn.cpt(dead)[:]=[[[0.9,0.1],[0.2,0.8]],[[0.89,0.11],[0.5,0.5]]]

['treatment', 'smoker', 'dead']


In [62]:
gnb.sideBySide(bn, bn.cpt("smoker"),bn.cpt("treatment"),bn.cpt("dead"))

### Question: How much will number of deads change if we decide treat everybody from now on with respect to not treating anyone?

\begin{align}
p(dead=1|do(treatment=1))=p(smoker=0)p(dead=1|treatment=1,smoker=0) + \\ p(smoker=1)p(dead=1|treatment=1,smoker=1)
\end{align}

In [71]:
expected_dead_if_treated = bn.cpt("smoker")[0]*bn.cpt("dead")[1,0,1] + bn.cpt("smoker")[1]*bn.cpt("dead")[1,1,1]
expected_dead_if_treated

0.22699999999999998

\begin{align}
p(dead=1|do(treatment=0))=p(smoker=0)p(dead=1|treatment=0,smoker=0) + \\ p(smoker=1)p(dead=1|treatment=0,smoker=1)
\end{align}

In [72]:
expected_dead_if_untreated = bn.cpt("smoker")[0]*bn.cpt("dead")[0,0,1] + bn.cpt("smoker")[1]*bn.cpt("dead")[0,1,1]
expected_dead_if_untreated

0.31

In [73]:
ATE = expected_dead_in_treated - expected_dead_in_untreated
ATE

-0.08300000000000002

### Response: The percentage of deads will decrease a 8.3% (it will go down from 31% to 22.7%). 
### Consequently, we SHOULD TREAT our population.

In [66]:
gen = gum.BNDatabaseGenerator(bn)

In [67]:
d = gen.drawSamples(1000000)

In [68]:
gen.toCSV("sample")

In [79]:
gum.fastBN('t->r<-x->t')

(gum::BayesNet<double>@0x55edc02a0950) BN{nodes: 3, arcs: 3, domainSize: 8, dim: 14}